_LowLevelCode/CPP/test/cpp_communicator.tests/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+set(TEST_NAME "cpp_communicator.test")
+
+set(TEST_SRC_FILES
+    "cpp_communicator_test.cpp"
+)
+
+set(SRC_FILES
+    "${CXX_SOURCE_DIR}/cpp_communicator/cpp_communicator.cpp"
+    "${CXX_SOURCE_DIR}/cpp_communicator/input_parser.cpp"
+    "${CXX_SOURCE_DIR}/cpp_communicator/MPI_wrapper.cpp"
+    "${CXX_SOURCE_DIR}/utility/environment.cpp"
+)
+
+set(HDR_FILES
+    "${CXX_SOURCE_DIR}/cpp_communicator/cpp_communicator.h"
+    "${CXX_SOURCE_DIR}/cpp_communicator/input_parser.h"
+    "${CXX_SOURCE_DIR}/cpp_communicator/MPI_wrapper.h"
+    "${CXX_SOURCE_DIR}/utility/environment.h"
+)
+include_directories(${MPI_CXX_HEADER_DIR})
```

Comment:
The tests which involve exception are failing as it does not serialize properly. The next ticket I am doing should fix this issue.

Reply:
`target_include_directories(${target} PRIVATE ${MPI_CXX_HEADER_DIR})` essentially just adds `-I /opt/mpi/include/` (or wherever the MPI include directory is) to the `g++` command for that target. 

Instead of `PRIVATE` you can specify `PUBLIC`. `PUBLIC` means that the headers will be part of the public interface of whatever you're building, private means you cannot access those headers through the library you're buillding. https://stackoverflow.com/questions/26243169/cmake-target-include-directories-meaning-of-scope

Using `include_directories` will add `-I /opt/mpi/include/` to __all__ targets that CMake builds, which we obviously don't want.